addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -283,6 +285,53 @@ var Wysiwyg = Widget.extend({
             mediaDialog.on('cancel', this, resolve);
         });
     },
+    changeSelectedTextColor(color, closeColorPicker = false) {
+        if(color === "") {
+            this.editor.execCommand('uncolorText');
+            $(window).trigger('click');
+            return;
+        }
+
+        if(this.colorPicker.colorNames.indexOf(color) !== -1) {
+           color = "var(--" + color + ")";
+        }
+        this.editor.execCommand('colorText', {color: color});
+
+        if(closeColorPicker) {
+            $(window).trigger('click');
+        }
+    },
+    toggleColorPicker() {
+        return new Promise((resolve)=> {
+            if(this.colorPicker) {
+                $(window).trigger('click');
+                return;
+            }
+            const self = this;
+            this.colorPicker = new ColorPaletteWidget(this, {});
+            this.colorPickerContainer = $('<div class="o_colorpicker_container"></div>');
+            this.colorPicker.appendTo(this.colorPickerContainer);
+
+            this.colorPicker.on('custom_color_picked', this,  async (e) => {
+                self.changeSelectedTextColor(e.data.color);
+            });
+            this.colorPicker.on('color_picked', this,  async (e) => {
+                self.changeSelectedTextColor(e.data.color, true);
+            });
+
+            this.$target.prepend(this.colorPickerContainer);
+            const closeColorPicker = function(e) {
+                if ($(e.target).parents('.o_colorpicker_container').length === 0) {
+                    self.colorPickerContainer.remove();
+                    self.colorPicker.destroy();
+                    self.colorPicker = undefined;
+                    $(window).off("click", closeColorPicker);
+                }
+            };
+
+            $(window).on("click", closeColorPicker);
```

Comment:
Maybe we can reuse the bootstrap dialog mechanism that odoo uses rather than creating a new one?

Reply:
Yes maybe, I'll have to check it then